ADAL/src/utils/ios/ADKeychainUtil.m

```diff
@@ -53,15 +53,16 @@ + (NSString*)retrieveTeamIDFromKeychain:(ADAuthenticationError * __autoreleasing
     NSDictionary *query = @{ (id)kSecClass : (id)kSecClassGenericPassword,
                              (id)kSecAttrAccount : @"teamIDHint",
                              (id)kSecAttrService : @"",
-                             (id)kSecAttrAccessible : (id)kSecAttrAccessibleAlways,
                              (id)kSecReturnAttributes : @YES };
     CFDictionaryRef result = nil;
     
     OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
     
     if (status == errSecItemNotFound)
     {
-        status = SecItemAdd((__bridge CFDictionaryRef)query, (CFTypeRef *)&result);
+        NSMutableDictionary* addQuery = [query mutableCopy];
+        [addQuery setObject:(id)kSecAttrAccessibleAlways forKey:(id)kSecAttrAccessible];
```

Comment:
That wouldn't hurt, but the same issue would be hit for current 2.2.2 apps. There could be devices with the teamIDHint value already created with kSecAttrAccessibleAlways. If the SecItemCopyMatching query contained kSecAttrAccessible==kSecAttrAccessibleAlwaysThisDeviceOnly, it wouldn't be able to find the old items with kSecAttrAccessible==kSecAttrAccessibleAlways. The safest change would be to leave the kSecAttrAccessible attribute out of the SecItemCopyMatching query and only put it in from the SecItemAdd query.